builder/builder-main.c

```diff
@@ -74,6 +75,7 @@ static GOptionEntry entries[] = {
   { "disable-download", 0, 0, G_OPTION_ARG_NONE, &opt_disable_download, "Don't download any new sources", NULL },
   { "disable-updates", 0, 0, G_OPTION_ARG_NONE, &opt_disable_updates, "Only download missing sources, never update to latest vcs version", NULL },
   { "download-only", 0, 0, G_OPTION_ARG_NONE, &opt_download_only, "Only download sources, don't build", NULL },
+  { "use-sources", 0, 0, G_OPTION_ARG_STRING_ARRAY, &opt_sources_dirs, "Specify a directory with sources to build from", "SOURCE-DIR"},
```

Comment:
Maybe add a note that you can specify this flag multiple times to add more directories? Unless this is somehow denoted in the usage info by the `G_OPTION_ARG_STRING_ARRAY` thing.